extensions/azurecore/src/azureResource/services/subscriptionService.ts

```diff
@@ -31,18 +31,25 @@ export class AzureResourceSubscriptionService implements IAzureResourceSubscript
 		for (const tenantId of tenantIds ?? account.properties.tenants.map(t => t.id)) {
 			try {
 				const token = await azdata.accounts.getAccountSecurityToken(account, tenantId, azdata.AzureResource.ResourceManagement);
-				const subClient = new SubscriptionClient(new TokenCredentials(token.token, token.tokenType), { baseUri: account.properties.providerSettings.settings.armResource.endpoint });
-				const newSubs = await subClient.subscriptions.list();
-				subscriptions.push(...newSubs.map(newSub => {
-					return {
-						id: newSub.subscriptionId,
-						name: newSub.displayName,
-						tenant: tenantId
-					};
-				}));
-				gotSubscriptions = true;
+				if (token !== undefined) {
```

Comment:
At the very least - it would be a good exercise to go through and check all uses of getAccountSecurityToken to ensure that they're handling undefined tokens correctly.

Reply:
I'll be doing more of token-related activity, wanted to keep this PR short to focus on the bug being fixed.

Reply:
Won't fix: To be done in a separate PR.